### PR TITLE
Close #6863 Add termly compatibility

### LINKED ISSUE #6863: Compatibility with Termly
**Is your feature request related to a problem? Please describe.**
[Termly](https://termly.io/) has an `auto blocking` mode that requires their script to be loaded early without any loading strategy applied. Otherwise, it will interfere with the plugin's functionality (intercept 3rd party scripts and decide if they can be executed or not depending on consent).

The indicator for this setting is stored on the options table under `termly_display_auto_blocker` set to `on`. 

The need is to exclude their main script `app.termly.io` (https://app.termly.io/resource-blocker/website-hash-identifier) from `defer` and `delay` when applicable.

**Describe the solution you'd like**
Have a compatibility file that will skip `defer` and `delay` when `termly_display_auto_blocker` is set to `on`

**Describe alternatives you've considered**
N/A

**Additional context**
Conversation with the Termly team - https://group-onecom.slack.com/archives/C07FUEY1939/p1723453296721739

### PR DESCRIPTION
**Pull request by Khadreal**
# Description
Exclude Termly from delay and defer when applicable.

Fixes #6863

## Type of change
- [x] Enhancement (non-breaking change which improves an existing functionality).

## Detailed scenario
Check #6863

### Documentation

Exclude Termly resources if `termly_display_auto_blocker` is set 'on' 

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

### CODE REVIEW

File: inc/ThirdParty/Plugins/Cookie/Termly.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	 * @return bool
+	 */
+	private function should_exclude(): bool {
+		$auto_block = get_option( 'termly_display_auto_blocker', 'off' );
+		if ( 'on' !== $auto_block ) {
+			return false;
+		}
+
+		return true;
+	}
```
Comment (jeawhanlee) [inline inc/ThirdParty/Plugins/Cookie/Termly.php]: I know the grooming required us to create a method like this, but seeing it now, it doesn't play a role, so I think we can send that logic to the method below.

File: inc/ThirdParty/Plugins/Cookie/Termly.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	/**
+	 * Excludes Termly JS files from delay JS
+	 *
+	 * @param array $exclude_delay_js Array of JS to be excluded.
+	 *
+	 * @return array
+	 */
+	public function exclude_defer_js( array $exclude_delay_js ): array {
```
Comment (jeawhanlee) [inline inc/ThirdParty/Plugins/Cookie/Termly.php]: This method name needs to change to something more generic and we need to update the docblock to be generic as well.

**Review by jeawhanlee — CHANGES_REQUESTED** [review]
It's good we have a unit test for this, but I think an integration test for this scenario would be helpful for testing with a html template containing that kind of link we are trying to exclude to see if the regex works properly.

**Comment by Mai-Saad** [comment]
@Khadreal Thanks for the PR. using the PR , can see that termly script is not excluded from deferjs /delay js
here https://new.rocketlabsqa.ovh/about. can you please check :pray: 
Screenshot from 2024-12-23 17-43-32
Screenshot from 2024-12-23 17-46-43
For test:
1- When termly has `termly_display_auto_blocker` on in options table => we exclude `https://app.termly.io/resource-blocker/` from defer js , delay js (try defer only on , delay only on , both on , all js modifications on) {it's already on in newlabs} => confirm that no console error/warning before/after interaction with the page
2- When termly option is off `termly_display_auto_blocker` , we aren't excluding the script from delay js / defer js

**Comment by Khadreal** [comment]
@Mai-Saad pushed a fix and tested that it's working now.

**Review by wordpressfan — APPROVED** [review]

**Review by hanna-meda — APPROVED** [review]
Thank you, @Khadreal.

After re-testing, I can confirm:
• We exclude https://app.termly.io/resource-blocker/ from defer and delay JS when `termly_display_auto_blocker` is enabled in the options table.
• We do not exclude https://app.termly.io/resource-blocker/ from defer and delay JS when `termly_display_auto_blocker` is disabled in the options table.

**Pull request merged**